src/ints/bios.cpp

```diff
@@ -1211,14 +1213,27 @@ class BIOS final : public Module_base{
 		/* INT 12 Memory Size default at 640 kb */
 		callback[2].Install(&INT12_Handler,CB_IRET,"Int 12 Memory");
 		callback[2].Set_RealVec(0x12);
-		if (IS_TANDY_ARCH) {
+		if (machine == MCH_TANDY) {
 			/* reduce reported memory size for the Tandy (32k graphics memory
 			   at the end of the conventional 640k) */
-			if (machine==MCH_TANDY) mem_writew(BIOS_MEMORY_SIZE,624);
-			else mem_writew(BIOS_MEMORY_SIZE,640);
-			mem_writew(BIOS_TRUE_MEMORY_SIZE,640);
-		} else mem_writew(BIOS_MEMORY_SIZE,640);
-		
+			mem_writew(BIOS_MEMORY_SIZE, 624);
+			mem_writew(BIOS_TRUE_MEMORY_SIZE, ConventionalMemorySizeKb);
+		} else if (machine == MCH_PCJR) {
+			const Section_prop* section = static_cast<Section_prop*>(control->GetSection("dos"));
+			assert(section);
+			const std::string pcjr_memory_config = section->Get_string("pcjr_memory_config");
+			if (pcjr_memory_config == "expanded") {
```

Comment:
It's unfortunate that both `dos_memory.cpp` and `bios.cpp` need to have knowledge about the setting strings.

A better solution would be that `dos_memory.cpp` does the parsing on module init, then converts it to an enum:

```cpp
enum class PcjrMemoryLayout { Standard, Expanded };
```

`bios.cpp` then could just ask for the memory layout with the public `DOS_GetPcjrMemoryLayout()` function.

I started using this more manageable approach in new code (e.g., `render.cpp`, `shader_manager.cpp`, etc.)

Reply:
That doesn't work.  BIOS init runs before DOS init does.  The config section that runs `BIOS_Init` is `[joystick]` for some reason and putting it there doesn't make much sense.

I get that it's nicer to work with enums but the config value is already parsed by the time we get here and in one of two valid states.  Introducing a global that depends on which module gets init'd first has been a major problem for us so I'd rather avoid that.

Reply:
Ok, let's leave it for now. When we refactor the config stuff, such enum conversions should happen before anything else is initialised anyway, so this would be just a stopgap measure until then.